samza-sql/src/test/java/org/apache/samza/sql/testutil/TestSourceResolverFactory.java

```diff
@@ -52,15 +159,29 @@ public SqlSystemSourceConfig fetchSourceInfo(String sourceName) {
         isTable = true;
         streamIdx = endIdx - 1;
       }
+
       Config systemConfigs = config.subset(sourceComponents[systemIdx] + ".");
       return new SqlSystemSourceConfig(sourceComponents[systemIdx], sourceComponents[streamIdx],
           Arrays.asList(sourceComponents), systemConfigs, isTable);
     }
 
     @Override
-    public boolean isTable(String sourceName) {
+    public boolean isTable(String sourceName){
       String[] sourceComponents = sourceName.split("\\.");
-      return sourceComponents[sourceComponents.length - 1].equalsIgnoreCase(SAMZA_SQL_QUERY_TABLE_KEYWORD);
+      return sourceComponents[sourceComponents.length - 1].equalsIgnoreCase(SAMZA_SQL_QUERY_TABLE_KEYWORD) ||
+          sourceName.equalsIgnoreCase("testDb.testTable");
```

Comment:
Why are we not using table keyword for output table ? If we do not use the table keyword, how do we plan to identify a table for a non-test source resolver ? IMO, we should not let test source resolver identify a table by <db>.<table>.

Reply:
Sorry, I did not think through this all the way when I commented earlier. Here is some context:
An input system in a given source resolver could be consumed either via a stream or a table. Hence, we added the table keyword to differentiate that. But for an output system, I think we made an explicit decision that an output system in a given source resolver would be only one of stream (SystemProducer) or table (TableProvider). So, what you are representing is correct. Could we explain the reasoning for this dichotomy between input and output systems in comments ? Of course, one could have their own source resolver to add table keyword even for output system.

Also, in this test resolver should we identify the table just by looking at the outputSystem component (could we rename testDb to testOutputSystem or some such ? testDb sounds like a db name) ?

Reply:
Thanks for the context. Regarding the naming though, here we're testing the ability to insert into a table so naming it testDb and testTable are clearer than calling it outputSystem, right?

Reply:
@pdu-mn1 @atoomula Since we have table keyword in the name as a convention in the input. I think it might be good idea to support the same convention in the output as well. It is ok if we don't need it. But it might be good idea to support it as well. What do you folks think?

i.e. we will support both

1. insert into <testDb.Table> 
2. insert into table.<testDb.table>

Does it make sense?